src/backend/core/api/serializers.py

```diff
@@ -426,8 +428,29 @@ def validate(self, attrs):
         attrs["expected_extension"] = self.context["expected_extension"]
         attrs["is_unsafe"] = self.context["is_unsafe"]
         attrs["content_type"] = self.context["content_type"]
+
+        if attrs["is_unsafe"]:
+            attrs["file"] = self.sanitize_file(attrs["file"], attrs["content_type"])
+
         return attrs
 
+    def sanitize_file(self, file_obj, mime_type):
+        """Sanitize the file content of markup mimetypes."""
+        if mime_type in [
+            "text/html",
+            "image/svg+xml",
+            "application/xhtml+xml",
+            "application/xml",
+        ]:
+            raw_content = file_obj.read()
+            decoded_str = raw_content.decode("utf-8", errors="replace")
+            sanitized_str = sanitize_markup(decoded_str)
+            sanitized_bytes = sanitized_str.encode("utf-8")
+            file_obj = ContentFile(sanitized_bytes, name=file_obj.name)
+
+        file_obj.seek(0)
+        return file_obj
```

Comment:
As it I don't like it. It will become a bottleneck if the file is large. You should test if beautifulsoup accepts in its constructor the `file_obj` directly, if it has the same signature as a python file object then it should handle it in a better way than using `read`

Reply:
As talked, commit removed.